internal/cluster/distribution/eks/eksprovider/workflow/create_infra.go

```diff
@@ -110,6 +110,25 @@ func CreateInfrastructureWorkflow(ctx workflow.Context, input CreateInfrastructu
 
 	ctx = workflow.WithActivityOptions(ctx, ao)
 
+	// create IAM role validate activity
+	var validateIAMRoleActivityFuture workflow.Future
+	if input.ClusterRoleID != "" {
+		{
+			activityInput := &ValidateIAMRoleActivityInput{
+				EKSActivityInput: commonActivityInput,
+				ClusterRoleID:    input.ClusterRoleID,
+			}
+			validateIAMRoleActivityFuture = workflow.ExecuteActivity(ctx, ValidateIAMRoleActivityName, activityInput)
+		}
+	}
+
+	if validateIAMRoleActivityFuture != nil {
+		validateIAMRoleActivityOutput := ValidateIAMRoleActivityOutput{}
+		if err := validateIAMRoleActivityFuture.Get(ctx, &validateIAMRoleActivityOutput); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
**Question**: do we benefit from using the future here instead of calling get on the execute call directly?
It's fine by me either way, just don't see much of a use for the async option.

Reply:
good point, I changed the code to calling it directly